Deprecation: use locationService in favor of getLocationSrv (#44813)
grafana/grafana: docs/sources/developers/plugins/migration-guide.md

- [From version 8.3.x to 8.4.x](#from-version-83x-to-84x)
  - [Value Mapping Editor has been removed from @grafana-ui library](#value-mapping-editor-has-been-removed-from-grafana-ui-library)
  - [Thresholds Editor has been removed from @grafana-ui library](#thresholds-editor-has-been-removed-from-grafana-ui-library)

>   - [8.3 Deprecations](#83-deprecations)

Reviewer: Should this be 8.4 as milestone and backport label?

Reviewer 2: ☝️

Author: Yes! Thanks for highlighting this 🙏🏻

Removed due to being an internal component.

> ### 8.3 deprecations

Reviewer: Should this be 8.4 to match milestone and backport?

> #### LocationService replaces getLocationSrv

Reviewer: Great work on these examples! 👏

> A while back we migrated to use a new routing system. At the same time we introduced a new service for managing locations, navigation etc. and we are now making that service the primary one to use.

Reviewer 3: ```suggestion
IIn a previous release, we migrated to use a new routing system and introduced a new service for managing locations, navigation, and related information. In this release, we are making that new service the primary service.  
```

> **Example:** How to import the service.

Reviewer 3: ```suggestion
**Example:** Import the service.
```

```ts
// before
import { getLocationSrv } from '@grafana/runtime';

// after
import { locationService } from '@grafana/runtime';
```

> **Example:** How to navigate to a path and add a new record in the navigation history so you can navigate back to the previous one.

Reviewer 3: ```suggestion
**Example:** Navigate to a path and add a new record in the navigation history so that you can navigate back to the previous one.
```

```ts
// before
getLocationSrv.update({
  path: '/route-to-navigate-to',
  replace: false,
});

// after
locationService.push('/route-to-navigate-to');
```

> **Example:** How to navigate to a path and replacing the current record in the navigation history.

Reviewer 3: ```suggestion
**Example:** Navigate to a path and replace the current record in the navigation history.
```

```ts
// before
getLocationSrv.update({
  path: '/route-to-navigate-to',
  replace: true,
});

// after
locationService.replace('/route-to-navigate-to');
```

> **Example:** How to update the search/query parameter for the current route and add a new record in the navigation history so you can navigate back to the previous one.

Reviewer 3: ```suggestion
**Example:** Update the search or query parameter for the current route and add a new record in the navigation history so that you can navigate back to the previous one.
```

```ts
// How to navigate to a new path
// before
getLocationSrv.update({
  query: {
    value: 1,
  },
  partial: true,
  replace: false,
});

// after
locationService.partial({ value: 1 });
```

> **Example:** How to update the search/query parameter for the current route and add replacing it in the navigation history.

Reviewer 3: ```suggestion
**Example:** Update the search or query parameter for the current route and add replacing it in the navigation history.
```
